---
title: "MarkerGenes"
author: "Chenxin Li"
date: "5/10/2022"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Pakages 
```{r}
library(tidyverse)
library(readxl)

library(RColorBrewer)
library(viridis)

library(svglite)

library(patchwork)
library(cowplot)
```
 
# Load orthogroups
```{r}
Orthogroups <- read_delim("../Results/OF_Results_Apr29/Orthologues/Orthologues_Araport_pep_20220119_representative_gene_model/Araport_pep_20220119_representative_gene_model__v__cro_v3.gene_models.repr.pep.tsv", 
                          delim = "\t", col_names = T, col_types = cols())

Orthogroups <- Orthogroups %>% 
  rename(
    Ath = Araport_pep_20220119_representative_gene_model, 
    Cro = cro_v3.gene_models.repr.pep
  )
head(Orthogroups)
```
# Load BLASTp
```{r}
BLASTp <- read_delim("../Results/Bash_Results/cro_v3_TAIR10_BLASTp_2022_05_10.out",
                     delim = "\t", col_names = F, col_types = cols())

head(BLASTp)
```
## Clean up blastp output 
```{r}
BLASTp_nice <- BLASTp %>% 
  arrange(X1) %>% 
  separate(X2, c("TAIR10", "TAIR10.isoform"), sep = "\\.") %>% 
  mutate(cro.v3 = str_remove(X1, "\\.\\d+")) %>% 
  group_by(cro.v3, TAIR10) %>% 
  slice_max(n = 1, order_by = X12) %>% 
  ungroup() %>% 
  filter(X11 < 1e-5) %>% 
  group_by(cro.v3) %>% 
  slice_max(n = 2, order_by = X12) %>% 
  ungroup()

head(BLASTp_nice)
```
 
# Load leaf markers
```{r}
AT_leaf_markers <- read_excel("../../JW_CRO/AT_leaf_markers.xlsx")

AT_leaf_markers_nice <- AT_leaf_markers %>% 
  gather("hits", "gene", hit1, hit2) %>% 
  mutate(gene = str_remove(gene, "T")) %>% 
  mutate(gene = str_replace(gene, "_", "-")) %>% 
  filter(is.na(gene) == F)
  
head(AT_leaf_markers_nice) 
```

## Match leaf marker to orthogroups 
```{r}
match_genes <- function(gene){
  Orthogroups %>% 
    filter(str_detect(Ath, as.character(gene))) 
}


leaf_marker_OG <- apply(X = AT_leaf_markers_nice[, 1], FUN = match_genes,  MARGIN = 1) %>% 
  bind_rows(.id = "index") %>% 
  full_join(
    AT_leaf_markers_nice %>% 
      cbind(index = 1:nrow(AT_leaf_markers_nice)) %>% 
      mutate(index = as.character(index)), by = "index"
  )  %>% 
  select(-hits, -gene, -index, -ref) %>% 
  inner_join(BLASTp_nice%>% 
               select(-TAIR10.isoform), by = c("TAIR"="TAIR10"))  

head(leaf_marker_OG, 60)
```

# Load root markers 
```{r}
AT_root_markers <- read_excel("../../JW_CRO/AT_root_marker.xlsx")

head(AT_root_markers)
```

## Match root markers to OG 
```{r}
root_marker_OG <- apply(X = AT_root_markers[, 1], FUN = match_genes,  MARGIN = 1) %>% 
  bind_rows(.id = "index") %>% 
  full_join(
    AT_root_markers %>% 
      cbind(index = 1:nrow(AT_root_markers)) %>% 
      mutate(index = as.character(index)), by = "index"
  )  %>% 
  select(-index) %>% 
  inner_join(BLASTp_nice %>% 
               select(-TAIR10.isoform), by = c("TAIR"="TAIR10"))

head(root_marker_OG, 60)
```
 



# Write out files 
```{r}
write_excel_csv(leaf_marker_OG, "../Results/R_outputs/leaf_markers.csv")
write_excel_csv(root_marker_OG, "../Results/R_outputs/root_markers.csv")
```



# MYC2
```{r}
Orthogroups %>% 
  filter(str_detect(Cro, "07G000280"))
```

