---
title: "Phase_HiCDC"
author: "Chenxin Li"
date: "3/8/2022"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Packages
```{r}
library(tidyverse)
library(readxl)

library(RColorBrewer)
library(viridis)

library(svglite)

library(patchwork)
library(cowplot)

library(HiCKeyR)
```

 
#TADS
## Quick look at the the sparse matrix generated by Straw 
```{r}
straw.out.list <- list.files(path = "../Results/Phase_20220308/straw_out/",
                             pattern = "Chr*", full.names = T)

count_data <- sapply(straw.out.list, read_delim, delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, col_types = cols(), simplify = F) %>% 
  bind_rows(.id = "id")

head(count_data)
```

```{r}
count_data_nice <- count_data %>% 
  mutate(Chr = str_remove_all(id, "../Results/Phase_20220308/straw_out//")) %>% 
  mutate(Chr = str_remove_all(Chr, ".txt"))

head(count_data_nice)
```
 

```{r}
count_data_nice %>% 
  filter(Chr == "Chr1") %>%
  filter(X1 < 2*10^7 & X1 > 10^7) %>% 
  filter(X2 < 2*10^7 & X2 > 10^7) %>% 
  mutate(count = case_when(
    X3 > 50 ~ 50,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_tile(aes(fill = count)) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       labels = c("0", "", ">50"),
                       breaks = c(1, 25, 50)) +
  labs(x = "Chr1: 10,000,000-20,000,000",
       y = "Chr1: 10,000,000-20.000,000") +
  theme_minimal() +
  theme(
    legend.position = c(0.8, 0.3),
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_blank(),
    panel.grid = element_blank()
  )
```

## Using HiCKeyR
```{r}
HiCKey_chr1 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr1.txt") 
HiCKey_chr2 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr2.txt") 
HiCKey_chr3 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr3.txt") 
HiCKey_chr4 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr4.txt") 
HiCKey_chr5 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr5.txt") 
HiCKey_chr6 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr6.txt") 
HiCKey_chr7 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr7.txt") 
HiCKey_chr8 <- segHeatMap("HiCKey-arg/arguments_HiCKey_CRO_AR_chr8.txt") 
```

## Check HiCKey output 
### Load data
```{r}
HiCKey_output_list <- list.files(path = "../Results/Phase_20220308/straw_out",
                                 pattern = "_output.txt", full.names = T)

HiCKey_out <- sapply(HiCKey_output_list, read_delim, delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, col_types = cols(), simplify = F) %>% 
  bind_rows(.id = "id")

head(HiCKey_out)
```

```{r}
bed_output_list <- list.files(path = "../Results/Phase_20220308/straw_out",
                                 pattern = "*.bed", full.names = T)

bed_out <- sapply(bed_output_list, read_delim, delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, col_types = cols(), simplify = F) %>% 
  bind_rows(.id = "id") %>% 
  distinct()

head(bed_out)
```

### Key data 
```{r}
HiCKey_out_nice <- HiCKey_out %>% 
  mutate(Chr = str_remove(id, "../Results/Phase_20220308/straw_out/")) %>% 
  mutate(Chr = str_remove(Chr, "_output.txt")) %>% 
  group_by(Chr) %>% 
  mutate(FDR = p.adjust(X3, "fdr")) %>% 
  filter(FDR < 0.05) %>% 
  mutate(FDR.text = case_when(
    FDR == 0 ~ "P < 2.2e-16",
    T ~ paste0("P = ", signif(FDR, 2))
  )) %>% 
  ungroup()

head(HiCKey_out_nice)
```
```{r}
bed_out_nice <- bed_out %>% 
  mutate(Chr = str_remove(id, "../Results/Phase_20220308/straw_out/")) %>% 
  mutate(Chr = str_remove(Chr, "_TADs.bed"))
  
head(bed_out_nice)
```

```{r}
HiCKey_out_nice %>% 
  filter(Chr == "chr1") %>% 
  filter(X1 < 5e6 & X1 > 4.5e6)  

bed_out_nice %>% 
  filter(Chr == "chr1") %>% 
  filter(X2 < 5e6 & X1 > 4.5e6) 
```
### make a graph

```{r}
bed_out_nice %>% 
  mutate(Chr = str_replace(Chr, "chr", "Chr")) %>% 
  mutate(hier = case_when(
    X3 > 5 ~ "fine",
    T ~ "coarse"
  )) %>% 
  filter(Chr == "Chr1") %>% 
  filter(X1 > 1e7 & X2 < 2e7) %>% 
  ggplot(aes(x = X1, y = X3)) +
  facet_grid(.~Chr, space = "free", scales = "free", switch = "x") +
  geom_rect(aes(xmin = X1, xmax = X2,
                ymin = X3 - 0.2, ymax = X3 + 0.2,
                fill = X3, color = hier),  size = 0.2)+
  scale_fill_viridis(option = "A",
                     breaks = c(2, 7),
                     labels = c("coarse", "fine")) +
  scale_color_manual(values = c("white", "black"),
                     limits = c("coarse", "fine")) +
  labs(x = "Chr1: 10,000,000 - 20,000,000",
       y = NULL,
       fill = "TAD hierarchy") +
  guides(color = "none") +
  theme_minimal() +
  theme(panel.grid = element_blank(),
        legend.position = "top",
        text = element_text(size = 14, color = "black", face = "bold"),
        axis.text = element_blank(),
        strip.text.x = element_blank())

#ggsave("../Results/Phase_20220308/R_outputs/TADs_linear.svg", height = 4, width = 7)
#ggsave("../Results/Phase_20220308/R_outputs/TADs_linear.png", height = 4, width = 7)
```


```{r}
count_data_nice %>% 
  filter(Chr == "Chr1") %>%
  filter(X1 < 5*10^6 & X1 > 4.5*10^6) %>% 
  filter(X2 < 5*10^6 & X2 > 4.5*10^6) %>% 
  mutate(count = case_when(
    X3 > 100 ~ 100,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_rect(data = bed_out_nice %>% 
    filter(Chr == "chr1") %>% 
    filter(X2 < 5e6 & X1 > 4.5e6),
            aes(xmin = X1, xmax = X2,
                ymin = X1, ymax = X2, 
                fill = X3), alpha = 0.1, color = NA) +
  scale_fill_gradientn(colors = viridis(10, option = "A")[2:8]) +
  guides(fill = "none") +
  ggnewscale::new_scale_fill() +
  geom_tile(aes(fill = count)) +
  geom_vline(data = HiCKey_out_nice %>%
               filter(Chr == "chr1") %>% 
               filter(X1 < 5e6 & X1 > 4.5e6),
             aes(xintercept = X1, alpha = 1/X2, color = X2),
             size = 1) +
  geom_hline(data = HiCKey_out_nice %>%
               filter(Chr == "chr1") %>% 
               filter(X1 < 5e6 & X1 > 4.5e6), 
             aes(yintercept = X1, alpha = 1/X2, color = X2),
             size = 1) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       labels = c("0", "50", ">100"),
                       breaks = c(1, 50, 100), na.value = "white") +
  scale_color_gradientn(colors = viridis(10, option = "A")[2:8],
                        breaks = c(1, 5),
                        labels = c("coarse", "\nfine")) +
  labs(x = "Chr1: 4,500,000 - 5,000,000",
       y = "Chr1: 4,500,000 - 5.000,000",
       color = "TAD\nhierarchy") +
  guides(alpha = "none",
         ) +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_blank(),
    panel.grid = element_blank()
  ) +
  coord_fixed()

#ggsave("../Results/Phase_20220308/R_outputs/Chr1_5M_TADs.svg", height = 5, width = 5)
#ggsave("../Results/Phase_20220308/R_outputs/Chr1_5M_TADs.png", height = 5, width = 5)
```

# Look at gene clustering 
## Load known genes
### v2 functional annotation 
```{r}
MIA_genes <- read_excel("~/Desktop/UGA/JW_CRO/MIA Gene List_Feb9_Sarah.xlsx")
head(MIA_genes)
```

```{r}
MIA_genes_root<- read_excel("~/Desktop/UGA/JW_CRO/MIA Gene List_root.xlsx")
head(MIA_genes_root)
```

## Load v3 gene coordinates 
```{r}
#v3 gff3
transcripts <- read_delim("../../CRO_v3/Data/cro_v3_anno/cro_v3.gene_models.repr.gff3",
                          col_names = F, delim = "\t", col_types = cols())

head(transcripts)
```
```{r}
genes <- transcripts %>% 
  filter(X3 == "mRNA") %>% 
  filter(str_detect(X1, "Chr")) %>% 
  separate(X9, c("ID", "Name"), sep = ";") %>% 
  mutate(gene = str_remove(Name, "Name=")) %>% 
  mutate(gene = str_sub(gene, 1, 13)) %>% 
  select(X1, X4, X5, X3, gene)  

head(genes)
```
### link v2 to v3 
```{r}
v3_curate <- read_excel("../../CRO_v3/Data/pathway_gene_v3.xlsx")
head(v3_curate)
```

```{r}
v3_curate_nice <- v3_curate %>% 
  mutate(cro_v2_id = str_remove(cro_v2_id, ".path1")) %>% 
  separate(cro_v3_region, c("Chr", "Start..End"), sep = ":") %>% 
  separate(Start..End, c("start", "end"), sep = "\\.\\.") %>% 
  mutate(start = as.numeric(start)) %>% 
  mutate(end = as.numeric(end)) %>% 
  mutate(gene = str_sub(cro_v3_gene, start = 1, end = 13))

head(v3_curate_nice)
```

```{r}
head(MIA_genes_root)
head(genes)

MIA_genes_info <- genes %>% 
  full_join(
    v3_curate_nice,
    by = "gene") %>% 
  full_join(
    MIA_genes_root %>% 
      rename(cro_v2_id = gene), by = c("cro_v2_id")
  ) %>% 
  filter(is.na(X1) == F) %>% 
  separate(annotation, c("tag", "function"), sep = "\\|")
  
MIA_genes_info  
```
```{r}
write_excel_csv(MIA_genes_info_root %>% 
                  filter(is.na(tag) == F), 
                file = "../Results/Phase_20220308/R_outputs/MIA_genes_info.csv")
```


## Visualization 
### MATE-TDC-STR cluster 

            
CRO_T125327	MATE 3	71684741	71691417	gene
CRO_T125328	TDC	3	71678061	71679563	gene
CRO_T125329	STR	3	71668329	71672111	gene
            MATE 3 71588447	71597028 
```{r}
HiCKey_out_nice %>%
               filter(X2 == 1) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 7.16e7 & X1 < 7.19e7) 
```

```{r}
TDC_clust1 <- count_data_nice %>% 
  filter(Chr == "Chr3") %>%
  filter(X1 > 7.16e7 & X1 < 7.19e7) %>% 
  filter(X2 > 7.16e7 & X2 < 7.19e7) %>% 
  mutate(count = case_when(
    X3 > 100 ~ 100,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_tile(aes(fill = count)) +
  geom_vline(data = HiCKey_out_nice %>%
               filter(X2 == 1) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 7.16e7 & X1 < 7.19e7),
             aes(xintercept = X1),
             size = 1, alpha = 0.8) +
  geom_hline(data = HiCKey_out_nice %>%
               filter(X2 == 1) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 7.16e7 & X1 < 7.19e7), 
             aes(yintercept = X1),
             size = 1, alpha = 0.8) +
  geom_text(
    data = HiCKey_out_nice %>%
               filter(X2 == 1) %>%
               filter(Chr == "chr3") %>%
               filter(X1 > 7.155e7 & X1 < 7.19e7),
             aes(label = FDR.text, x = X1, y = -Inf),
             size = 3, fontface = "bold", vjust = -0.3, hjust = 1.1,
  ) +
  #geom_vline(xintercept = 71668329, size = 0.8, linetype = 4) +
  #geom_vline(xintercept = 71691417, size = 0.8, linetype = 4) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       #colors = viridis(n = 10, option = "A"),
                       labels = c("0", "50", ">100"),
                       breaks = c(1, 50, 100)) +
  # scale_color_gradientn(colors = viridis(10, option = "A")[2:8],
  #                       breaks = c(1,4), 
  #                       labels = c("coarse", "\nfine")) +
  xlim(c(7.16e7, 7.19e7)) +
  ylim(c(7.16e7, 7.19e7)) +
  labs(x = NULL,
       y = "Chr3\n71,600,000-71,900,000",
       fill = "Contacts") +
  guides(alpha = "none",
         ) +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_blank(),
    panel.grid = element_blank(),
    legend.position = "none"
  )   

TDC_clust1
```

```{r}
TDC_gene_track <- MIA_genes_info %>% 
  filter(X1 == "Chr3") %>% 
  filter(X4 > 7.16e7 & X5 < 7.19e7) %>%
  mutate(tag2 = case_when(
    is.na(tag) ~ "other",
    tag == "MATE" ~ "SLTr",
    T ~ tag
  )) %>% 
  ggplot(aes(x = X4, y = 1)) +
  geom_hline(yintercept = 1, size = 1) +
  geom_rect(aes(xmin = X4, xmax = X5,
                fill = tag2),
            ymin = 0.75, ymax = 1.25) +
  #geom_vline(xintercept = 71668329, size = 0.8, linetype = 4) +
  #geom_vline(xintercept = 71691417, size = 0.8, linetype = 4) +
  scale_fill_manual(values = c(brewer.pal(8, "Accent")[c(1,3,5)], "grey80"),
                    limits = c("STR", "TDC", "SLTr", "other")) +
  labs(x = NULL,
       y = NULL,
       fill = "Genes") +
  guides(fill = guide_legend(nrow = 2)) +
  xlim(c(7.16e7, 7.19e7)) +
  theme_minimal() +
  theme(
    panel.grid = element_blank(),
    text = element_text(size = 14, color = "black", face = "bold"),
    axis.text.y = element_blank(),
    axis.text.x = element_blank(),
    legend.position = "bottom"
  )

TDC_gene_track
```

```{r}
plot_grid(TDC_clust1, TDC_gene_track,
          nrow = 2, align = "v", axis = "lr",
          rel_heights = c(0.7, 0.35))

ggsave("../Results/Phase_20220308/R_outputs/STR_cluster.svg", 
       height = 3.5, width = 2.8, bg = "white")
ggsave("../Results/Phase_20220308/R_outputs/STR_cluster.png", 
       height = 3.5, width = 2.8, bg = "white")
```

```{r}
MIA_genes_info %>% 
  filter(X1 == "Chr3") %>% 
  filter(X4 > 7.16e7 & X5 < 7.19e7)  %>% 
  arrange(X4) 
```


### T16H2 and 16OMT
CRO_T110598	T16H2	3	69980470	69982789	
CRO_T110596	16OMT 3	69995704	69997603	
Two other CYPs in this cluster 
 
left: 69980470
rigth: 70018608 
 


```{r}
HiCKey_out_nice %>%
              # filter(X2 == 4) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 6.99e7 & X1 < 7.002e7) 
```


 
```{r}
t16_clust1 <- count_data_nice %>% 
  filter(Chr == "Chr3") %>%
  filter(X1 > 6.99e7 & X1 < 7.002e7) %>% 
  filter(X2 > 6.99e7 & X2 < 7.002e7) %>% 
  mutate(count = case_when(
    X3 > 100 ~ 100,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_tile(aes(fill = count)) +
  geom_vline(data = HiCKey_out_nice %>%
            #   filter(X2 == 4) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 6.99e7 & X1 < 7.002e7),
             aes(xintercept = X1),
             size = 1, alpha = 0.8) +
  geom_hline(data = HiCKey_out_nice %>%
             #  filter(X2 == 4) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 6.99e7 & X1 < 7.002e7), 
             aes(yintercept = X1),
             size = 1, alpha = 0.8) +
 # geom_vline(xintercept = 69980470, size = 0.8, linetype = 4) +
 # geom_vline(xintercept = 70018608, size = 0.8, linetype = 4) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       #colors = viridis(n = 10, option = "A"),
                       labels = c("0", "50", ">100"),
                       breaks = c(1, 50, 100)) +
  # scale_color_gradientn(colors = viridis(10, option = "A")[2:8],
  #                       breaks = c(4, 7),
  #                       labels = c("coarse", "fine")
  #                      ) +
  xlim(c(6.99e7, 7.002e7)) +
  ylim(c(6.99e7, 7.002e7)) +
  labs(x = NULL,
       y = "Chr3: 69,900,000 - 70,020,000",
       fill = "Contacts") +
  guides(alpha = "none",
         ) +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", color = "black"),
    legend.position = "none",
    axis.text = element_blank(),
    panel.grid = element_blank()
  )   

t16_clust1
```
```{r}
T16_gene_track <- MIA_genes_info %>% 
  filter(X1 == "Chr3") %>% 
  filter(X4 > 6.99e7 & X5 < 7.002e7) %>% 
  mutate(tag2 = case_when(
    is.na(tag) ~ "other",
    T ~ tag
  )) %>% 
  ggplot(aes(x = X4, y = 1)) +
  geom_hline(yintercept = 1, size = 1) +
  geom_rect(aes(xmin = X4, xmax = X5,
                fill = tag2),
            ymin = 0.75, ymax = 1.25) +
  #geom_vline(xintercept = 69980470, size = 0.8, linetype = 4) +
  #geom_vline(xintercept = 70768219, size = 0.8, linetype = 4) +
 #scale_fill_manual(values = c(brewer.pal(8, "Accent")[c(1, 3, 5)], "grey80"),
   #                 limits = c("T16H2", "16OMT", "GES", "other")) +
  labs(x = NULL,
       y = NULL,
       fill = "Genes") +
  xlim(c(6.99e7, 7.002e7)) +
  theme_minimal() +
  theme(
    panel.grid = element_blank(),
    text = element_text(size = 14, color = "black", face = "bold"),
    axis.text.y = element_blank(),
    axis.text.x = element_text(color = "black"),
    legend.position = "bottom"
  )

T16_gene_track
```
```{r}
plot_grid(t16_clust1, T16_gene_track,
          nrow = 2, align = "v", axis = "lr",
          rel_heights = c(1, 0.3))

ggsave("../Results/Phase_20220308/R_outputs/T16_cluster.svg", height = 5, width = 5.1)
ggsave("../Results/Phase_20220308/R_outputs/T16_cluster.png", height = 5, width = 5.1)
```

```{r}
MIA_genes_info %>% 
  filter(X1 == "Chr3") %>% 
  filter(X4 > 6.99e7 & X5 < 7.002e7) %>% 
  arrange(X4)
```
Checked annotation, the gene before, in between, and after T16H2/T16OMT are all CYPs. 

### SGDs
CRO_T128799	SGD 3	4240485	4248047	
CRO_T111320	SGD 3	4297272	4302781	

left = 4,240,485
right = 4,302,781

```{r}
SDG_clust1 <- count_data_nice %>% 
  filter(Chr == "Chr3") %>%
  filter(X1 > 4.1e6 & X1 < 4.5e6) %>% 
  filter(X2 > 4.1e6 & X2 < 4.5e6) %>% 
  mutate(count = case_when(
    X3 > 100 ~ 100,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_tile(aes(fill = count)) +
  geom_vline(data = HiCKey_out_nice %>%
               filter(X2 == 1) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 4.1e6 & X1 < 4.5e6),
             aes(xintercept = X1),
             size = 1, alpha = 0.8) +
  geom_hline(data = HiCKey_out_nice %>%
               filter(X2 == 1) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 4.1e6 & X1 < 4.5e6), 
             aes(yintercept = X1),
             size = 1, alpha = 0.8) +
  geom_text(
    data = HiCKey_out_nice %>%
               filter(X2 == 1) %>% 
               filter(Chr == "chr3") %>% 
               filter(X1 > 4.1e6 & X1 < 4.5e6),
             aes(label = FDR.text, x = Inf, y = X1),
             size = 3, fontface = "bold", vjust = -0.3, hjust = 1
  ) +
  geom_vline(xintercept = 4217156, size = 0.8, linetype = 4) +
  geom_vline(xintercept = 4302781, size = 0.8, linetype = 4) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       #colors = viridis(n = 10, option = "A"),
                       labels = c("0", "50", ">100"),
                       breaks = c(1, 50, 100)) +
  # scale_color_gradientn(colors = viridis(10, option = "A")[2:8],
  #                       breaks = c(1, 4),
  #                       labels = c("coarse", "fine")) +
  xlim(c(4.1e6, 4.5e6)) +
  ylim(c(4.1e6, 4.5e6)) +
  labs(x = NULL,
       y = "Chr3\n4,100,000-4,500,000",
       fill = "Contacts") +
  guides(alpha = "none",
         ) +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_blank(),
    panel.grid = element_blank(),
    legend.position = "none"
  )   

SDG_clust1
```
```{r}
SDG_gene_track <- MIA_genes_info %>% 
  filter(X1 == "Chr3") %>% 
  filter(X4 > 4.1e6 & X5 < 4.5e6) %>%
  mutate(tag2 = case_when(
    is.na(tag) ~ "other",
    T ~ tag
  )) %>% 
  mutate(tag2 = case_when(
    str_detect(tag2, "SGD$") ~ "SGD1",
    #str_detect(tag2, "SGD2") ~ "SDG2",
    str_detect(gene, "03G003540") ~ "SS1",
    str_detect(gene, "03G003550") ~ "SS2",
    T ~ tag2
  )) %>% 
  ggplot(aes(x = X4, y = 1)) +
  geom_hline(yintercept = 1, size = 1) +
  geom_rect(aes(xmin = X4, xmax = X5,
                fill = tag2),
            ymin = 0.75, ymax = 1.25) +
  #geom_vline(xintercept = 4217156, size = 0.8, linetype = 4) +
  #geom_vline(xintercept = 4302781, size = 0.8, linetype = 4) +
  scale_fill_manual(values = c(brewer.pal(8, "Accent")[c(1,3,5,6)], "grey80"),
                    limits = c("SS1", "SS2", "SGD1", "SGD2", "other")) + # fix labels later
  labs(x = NULL,
       y = NULL,
       fill = "Genes") +
  guides(fill = guide_legend(nrow = 3)) +
  xlim(c(4.1e6, 4.5e6)) +
  theme_minimal() +
  theme(
    panel.grid = element_blank(),
    text = element_text(size = 14, color = "black", face = "bold"),
    axis.text.y = element_blank(),
    axis.text.x = element_blank(),
    legend.position = "bottom"
  )

SDG_gene_track
```

```{r}
plot_grid(SDG_clust1, SDG_gene_track,
          nrow = 2, align = "v", axis = "lr",
          rel_heights = c(0.75, 0.4))

ggsave("../Results/Phase_20220308/R_outputs/SDG_cluster.svg", 
       height = 4, width = 3, bg = "white")
ggsave("../Results/Phase_20220308/R_outputs/SDG_cluster.png", 
       height = 4, width = 3, bg = "white")
```

```{r}
MIA_genes_info %>% 
  filter(X1 == "Chr3") %>% 
  filter(X4 > 4.1e6 & X5 < 4.5e6) %>% 
  arrange(X4)
```
The gene, CRO_T111319, right before SDG is also a glucosidase.
CRO_T128797, one of the genes between them is "oligopeptide transporter".


### DAT, MAT and TAT 
CRO_T120021	DAT 	2	1605943	1607262
CRO_T120028	MAT 	2	1548774	1550105	
CRO_T120026	TAT	 	2	1561363	1562673

left: 1,400,000
right: 1,650,000

```{r}
AT_clust1 <- count_data_nice %>% 
  filter(Chr == "Chr2") %>%
  filter(X1 > 1.435e6 & X1 < 1.65e6) %>% 
  filter(X2 > 1.435e6 & X2 < 1.65e6) %>% 
  mutate(count = case_when(
    X3 > 100 ~ 100,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_tile(aes(fill = count)) +
  geom_vline(data = HiCKey_out_nice %>%
               filter(X2 <= 2) %>% 
               filter(Chr == "chr2") %>%
               filter(X1 > 1.435e6 & X1 < 1.65e6),
             aes(xintercept = X1),
             size = 1, alpha = 0.8) +
  geom_hline(data = HiCKey_out_nice %>%
               filter(X2 <= 2) %>% 
               filter(Chr == "chr2") %>%
               filter(X1 > 1.435e6 & X1 < 1.65e6),
             aes(yintercept = X1),
             size = 1, alpha = 0.8) +
  geom_text(
    data = HiCKey_out_nice %>%
               filter(X2 <= 2) %>% 
               filter(Chr == "chr2") %>% 
               filter(X1 > 1.435e6 & X1 < 1.65e6),
             aes(label = FDR.text, x = Inf, y = X1),
             size = 3, fontface = "bold", vjust = -0.2, hjust = 1,
  ) +
  #geom_vline(xintercept = 1548774, size = 0.8, linetype = 4) + #MAT start
  #geom_vline(xintercept = 1562673, size = 0.8, linetype = 4) + #TAT end 
  #geom_vline(xintercept = 1605943, size = 0.8, linetype = 4) + #DAT start
  #geom_vline(xintercept = 1607262, size = 0.8, linetype = 4) + #DAT end 
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       #colors = viridis(n = 10, option = "A"),
                       labels = c("0", "50", ">100"),
                       breaks = c(1, 50, 100)) +
  # scale_color_gradientn(colors = viridis(10, option = "A")[2:8],
  #                       breaks = c(1, 2),
  #                       labels = c("coarse", "fine")
  #                      ) +
 xlim(c(1.435e6, 1.65e6)) +
 ylim(c(1.435e6, 1.65e6)) +
  labs(x = NULL,
       y = "Chr2\n1,435,000-1,650,000",
       fill = "Contacts") +
  guides(alpha = "none",
         ) +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_blank(),
    panel.grid = element_blank(),
    legend.position = "none"
  )   

AT_clust1
```
```{r}
DAT_gene_track <- MIA_genes_info %>% 
  filter(X1 == "Chr2") %>% 
  filter(X4 > 1.45e6 & X5 < 1.65e6) %>% 
  mutate(tag2 = case_when(
    is.na(tag) == F ~ tag,
    T ~ "other"
  )) %>% 
  ggplot(aes(x = X4, y = 1)) +
  geom_hline(yintercept = 1, size = 1) +
  geom_rect(aes(xmin = X4, xmax = X5,
                fill = tag2),
            ymin = 0.75, ymax = 1.25) +
  #geom_vline(xintercept = 1548774, size = 0.8, linetype = 4) + #MAT start
  #geom_vline(xintercept = 1562673, size = 0.8, linetype = 4) + #TAT end 
  #geom_vline(xintercept = 1605943, size = 0.8, linetype = 4) + #DAT start
  #geom_vline(xintercept = 1607262, size = 0.8, linetype = 4) + #DAT end 
 scale_fill_manual(values = c(brewer.pal(8, "Accent")[c(1,3,5)], "grey80"),
                   limits = c("MAT", "TAT", "DAT", "other")) +
  labs(x = NULL,
       y = NULL,
       fill = "Genes") +
  guides(fill = guide_legend(nrow = 2)) +
  xlim(c(1.435e6, 1.65e6)) +
  theme_minimal() +
  theme(
    panel.grid = element_blank(),
    text = element_text(size = 14, color = "black", face = "bold"),
    axis.text.y = element_blank(),
    axis.text.x = element_blank(),
    legend.position = "bottom"
  )

DAT_gene_track
```

```{r}
plot_grid(AT_clust1, DAT_gene_track,
          nrow = 2, align = "v", axis = "lr",
          rel_heights = c(0.7, 0.35))

ggsave("../Results/Phase_20220308/R_outputs/DAT_cluster.svg", 
       height = 3.5, width = 2.8, bg = "white")
ggsave("../Results/Phase_20220308/R_outputs/DAT_cluster.png", 
       height = 3.5, width = 2.8, bg = "white")
```


```{r}
MIA_genes_info %>% 
  filter(X1 == "Chr2") %>% 
  filter(X4 > 1.45e6 & X5 < 1.65e6) %>% 
  arrange(X4)  
```

# Loops
```{r}
CRO_AR_loops <- read_delim("../Results/Phase_20220308/Phase_hiccups/merged_loops.bedpe", 
    col_names = FALSE, skip = 2, col_types = cols(), delim = "\t")

head(CRO_AR_loops)
```
Hiccup output headers: 
1chromosome1    2x1    3x2    
4chromosome2    5y1    6y2    
11color    12observed
13expected_bottom_left    14expected_donut    15expected_horizontal    16expected_vertical    
17fdr_bottom_left    18fdr_donut    19fdr_horizontal    20fdr_vertical    
21number_collapsed   22centroid1    23centroid2    24radius

## Filtering 
Only keep Chr 1-8 
```{r}
CRO_AR_loops_filter <- CRO_AR_loops %>% 
  mutate(Chr.from = str_remove(X1, "HiC_scaffold_") %>% as.numeric()) %>% 
  mutate(Chr.to = str_remove(X4, "HiC_scaffold_") %>% as.numeric()) %>% 
  filter(Chr.from <= 8) %>% 
  filter(Chr.to <= 8)

dim(CRO_AR_loops_filter)
head(CRO_AR_loops_filter)
```

```{r}
CRO_AR_loops_filter %>% 
  summarise(cutoff.17 = max(X17),
            cutoff.18 = max(X18),
            cutoff.19 = max(X19),
            cutoff.20 = max(X20))
```


```{r}
CRO_AR_loops_filter %>% 
  mutate(distance = X5-X2) %>% 
  ggplot(aes(x = X15, y = X12)) + 
  geom_point(alpha = 0.5, aes(fill = -log10(X20+10^-30)),
             shape = 21, color = "black") +
  geom_abline(slope = 1, intercept = 0) +
  scale_fill_viridis(option = "D")+
  labs(x = "Expected",
       y = "Observed",
       fill = "-log10(FDR)") +
  theme_classic()

ggsave("../Results/Phase_20220308/R_outputs/Obs_vs_Exp.svg", height = 5, width = 5)
ggsave("../Results/Phase_20220308/R_outputs/Obs_vs_Exp.png", height = 5, width = 5)
```

```{r}

CRO_AR_loops_filter %>% 
  mutate(distance = (X5-X2)/1000) %>% 
  ggplot(aes(x = distance, y = X15)) + 
  geom_point(alpha = 0.5,
             shape = 21, color = "white", fill = "grey20") +
  scale_x_log10() +
  scale_fill_viridis(option = "D")+
  labs(x = "Distance (kb)",
       y = "Expected",
       fill = "-log10(FDR)") +
  theme_classic()

ggsave("../Results/Phase_20220308/R_outputs/Exp_vs_dist.svg", height = 5, width = 5)
ggsave("../Results/Phase_20220308/R_outputs/Exp_vs_dist.png", height = 5, width = 5)
```

## check salient features
### spot at Chr 3 
```{r}
CRO_AR_loops_filter %>% 
  #head() %>% 
  filter(X1 == "HiC_scaffold_3") %>% 
  filter(X4 == "HiC_scaffold_3") %>% 
  filter(X2 > 4.5*10^7 & X2 < 5.1*10^7) %>% 
  arrange(X2)
```
```{r}
count_data_nice %>% 
  filter(Chr == "Chr3") %>%
  filter(X1 > 4.8*10^7 & X1 < 6.4*10^7) %>% 
  filter(X2 > 4.8*10^7 & X2 < 6.4*10^7) %>% 
  mutate(count = case_when(
    X3 > 10 ~ 10,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_tile(aes(fill = count)) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       labels = c("0", "5", ">10"),
                       breaks = c(1, 5, 10), na.value = "white") +
  geom_vline(xintercept = 4.99e7, alpha = 0.8) +
  geom_vline(xintercept = 5.1e7, alpha = 0.8) +
  geom_hline(yintercept = 6.2e7, alpha = 0.8) +
  geom_hline(yintercept = 6.26e7, alpha = 0.8) +
  labs(x = "Chr3: 48,000,000 - 64,000,000",
       y = "Chr3: 48,000,000 - 64,000,000",
       fill = "Contacts") +
  guides(alpha = "none",
         ) +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_blank(),
    panel.grid = element_blank()
  ) +
  coord_fixed()
```
 

## write out for bedtools nearest 
```{r}
CRO_AR_loops_filter_x <- CRO_AR_loops_filter %>% 
  mutate(tag = paste0(X1, ":", X2, "-", X4, ":", X5)) %>% 
  select(X1, X2, X3, tag) %>% 
  arrange(X1, X2, X3)

head(CRO_AR_loops_filter_x)

CRO_AR_loops_filter_y <- CRO_AR_loops_filter %>% 
  mutate(tag = paste0(X1, ":", X2, "-", X4, ":", X5)) %>% 
  select(X4, X5, X6, tag) %>% 
  arrange(X4, X5, X6)

head(CRO_AR_loops_filter_y)
```

```{r}
write_delim(CRO_AR_loops_filter_x, 
            "../Results/Phase_20220308/R_outputs/CRO_AR_loops_x.bed",
            col_names = F, delim = "\t")

write_delim(CRO_AR_loops_filter_y, 
            "../Results/Phase_20220308/R_outputs/CRO_AR_loops_y.bed",
            col_names = F, delim = "\t")
```

## Load Bedtools outputs
What are genes close to or overlapping loops?
```{r}
loop_x_close <- read_delim("../Results/Phase_20220308/R_outputs/CRO_AR_loops_x_genes.txt",
                           delim = "\t", col_names = F, col_types = cols())

loop_y_close <- read_delim("../Results/Phase_20220308/R_outputs/CRO_AR_loops_y_genes.txt",
                           delim = "\t", col_names = F, col_types = cols())

head(loop_x_close)
```
## Wrangle data 
```{r}
colnames(loop_x_close) <- c(
  "Chr.from", "loop.start.from", "loop.end.from", "tag",
  "X5", "gene.start.from", "gene.end.from", "gene.from"
)

colnames(loop_y_close) <- c(
  "Chr.to", "loop.start.to", "loop.end.to", "tag",
  "X5", "gene.start.to", "gene.end.to", "gene.to"
)
```

```{r}
loop_gene_info <- loop_x_close %>% 
  select(-X5) %>% 
  inner_join(
    loop_y_close %>% 
      select(-X5),
    by = "tag"
  ) %>% 
  mutate(distance.x.1 = abs(
    loop.start.from + 2500 - gene.start.from
  )) %>% 
    mutate(distance.y.1 = abs(
    loop.start.to + 2500 - gene.start.to
  )) %>% 
  mutate(distance.x.2 = abs(
    loop.start.from + 2500 - gene.end.from
  )) %>% 
    mutate(distance.y.2 = abs(
    loop.start.to + 2500 - gene.end.to
  ))  %>% 
  mutate(distance.x = case_when(
    distance.x.1 <= distance.x.2 ~ distance.x.1,
    T ~ distance.x.2
  )) %>% 
   mutate(distance.y = case_when(
    distance.y.1 <= distance.y.2 ~ distance.y.1,
    T ~ distance.y.2
  ))  

head(loop_gene_info)
```

```{r}
# loop_gene_info %>% 
#   gather("dim", "dist", distance.x, distance.y) %>% 
#   ggplot(aes(x = dist)) +
#   facet_grid(dim ~.) +
#   geom_histogram(binwidth = 100) +
#   theme_bw()

summary(loop_gene_info$distance.x)
summary(loop_gene_info$distance.y)
```
Will use 5-kb cutoff for loop-associated genes - that's the resolution of loop finder.  


```{r}
loop_gene_info_MIA <- loop_gene_info %>% 
  filter(gene.from %in% MIA_genes$gene |
           gene.to %in% MIA_genes$gene) 

loop_gene_info_MIA
```
```{r}
MIA_genes_info %>% 
  filter(cro_v2_id %in% loop_gene_info_MIA$gene.from |
           cro_v2_id %in% loop_gene_info_MIA$gene.to)
```

## Visualize 
### PAS-THAS

HiC_scaffold_6	58965000	58970000		
58962146	58964495	CRO_T113150
59020000	59025000	59017520	59019109	CRO_T113148

left: 58962146
right: 59019109

```{r}
THAS_loop1 <- count_data_nice %>% 
  filter(Chr == "Chr6") %>%
  filter(X1 > 5.89e7 & X1 < 5.91e7) %>% 
  filter(X2 > 5.89e7 & X2 < 5.91e7) %>% 
  mutate(count = case_when(
    X3 > 100 ~ 100,
    T ~ X3
  )) %>% 
  ggplot(aes(x = X1, y = X2)) +
  geom_tile(aes(fill = count)) +
  geom_vline(data = HiCKey_out_nice %>%
               filter(X2 <= 3) %>% 
               filter(Chr == "chr6") %>%
               filter(X1 > 5.89e7 & X1 < 5.91e7),
             aes(xintercept = X1),
             size = 1, alpha = 0.8) +
  geom_hline(data = HiCKey_out_nice %>%
               filter(X2 <= 3) %>% 
               filter(Chr == "chr6") %>%
               filter(X1 > 5.89e7 & X1 < 5.91e7),
             aes(yintercept = X1),
             size = 1, alpha = 0.8) +
  geom_text(
    data = HiCKey_out_nice %>%
               filter(X2 <= 3) %>% 
               filter(Chr == "chr6") %>% 
               filter(X1 > 5.89e7 & X1 < 5.91e7), 
             aes(label = FDR.text, x = Inf, y = X1),
             size = 3, fontface = "bold", vjust = -0.2, hjust = 1
  ) +
  # geom_vline(xintercept = 58962146, size = 0.8, linetype = 4) +
  # geom_vline(xintercept = 59019109, size = 0.8, linetype = 4) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       #colors = viridis(n = 10, option = "A"),
                       limits = c(0, 100),
                       labels = c("0", "50", ">100"),
                       breaks = c(1, 50, 100)) +
  # scale_color_gradientn(colors = viridis(10, option = "A")[2:8],
  #                       breaks = c(2, 4),
  #                       labels = c("coarse", "fine")
  #                       ) +
  xlim(c(5.89e7, 5.91e7)) +
  ylim(c(5.89e7, 5.91e7)) +
  labs(x = NULL,
       y = "Chr6\n58,900,000-59,100,000",
       fill = "Contacts") +
  guides(alpha = "none",
         ) +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_blank(),
    panel.grid = element_blank(),
    legend.position = "none"
  )   

THAS_loop1
```

 

```{r}
THAS_gene_track <- MIA_genes_info %>% 
  filter(X1 == "Chr6") %>% 
  filter(X4 > 5.89e7 & X5 < 5.91e7) %>% 
  mutate(tag2 = case_when(
    is.na(tag) ~ "other",
    str_detect(cro_v2_id, "CRO_T113150") ~ "THAS2",
    str_detect(cro_v2_id, "CRO_T113153") ~ "GS2",
    str_detect(cro_v2_id, "CRO_T113154") ~ "GS1",
    T ~ tag
  )) %>% 
  ggplot(aes(x = X4, y = 1)) +
  geom_hline(yintercept = 1, size = 1) +
  geom_rect(aes(xmin = X4, xmax = X5,
                fill = tag2),
            ymin = 0.75, ymax = 1.25) +
  geom_curve(
    data = loop_gene_info_MIA,
    aes(x = loop.start.from+2500, xend = loop.start.to+2500),
    y = 1, yend = 1, size = 1, curvature = -0.25, ncp = 100
  ) +
  scale_fill_manual(values = c(brewer.pal(8, "Accent")[c(1,3,5,6)], "grey80"),
                    limits = c( "GS1", "GS2", "THAS2", "PAS", "other")) +
  labs(x = NULL,
       y = NULL,
       fill = "Genes") +
  guides(fill = guide_legend(nrow = 3)) +
  xlim(c(5.89e7, 5.91e7)) +
  theme_minimal() +
  theme(
    panel.grid = element_blank(),
    text = element_text(size = 14, color = "black", face = "bold"),
    axis.text.y = element_blank(),
    axis.text.x = element_blank(),
    legend.position = "bottom"
  )

THAS_gene_track
```
```{r}
plot_grid(THAS_loop1, THAS_gene_track,
          nrow = 2, align = "v", axis = "lr",
          rel_heights = c(0.75, 0.4))

ggsave("../Results/Phase_20220308/R_outputs/THAS_loop.svg", 
       height = 4, width = 3, bg = "white")
ggsave("../Results/Phase_20220308/R_outputs/THAS_loop.png", height = 4, width = 3,
       bg = "white")
```
 

```{r}
MIA_genes_info %>% 
  filter(X1 == "Chr6") %>% 
  filter(X4 > 5.89e7 & X5 < 5.91e7) %>% 
  mutate(tag = case_when(
    str_detect(cro_v2_id, "CRO_T113150") ~ "THAS",
    T ~ tag
  )) %>% 
  arrange(X4)
```
 


  
 